src/vs/editor/contrib/stickyScroll/browser/stickyScroll.ts

```diff
@@ -0,0 +1,338 @@
+	}
+
+	dispose(): void {
+		this._store.dispose();
+		this._sessionStore.dispose();
+	}
+}
+
+const _ttPolicy = window.trustedTypes?.createPolicy('stickyScrollViewLayer', { createHTML: value => value });
+
+class StickyScrollCodeLine {
+	constructor(private readonly _line: string, private readonly _lineNumber: number, private readonly _editor: IActiveCodeEditor,
+		private readonly _zIndex: number, private readonly _position?: number) { }
+
+
+	getDomNode() {
+
+		const root: HTMLElement = document.createElement('div');
+		const modifiedLine = this._line.replace(/\s/g, '\xa0');
+		const lineRenderingData = this._editor._getViewModel().getViewLineRenderingData(this._editor.getVisibleRangesPlusViewportAboveBelow()[0], this._lineNumber);
+		const actualInlineDecorations = LineDecoration.filter(lineRenderingData.inlineDecorations, this._lineNumber, lineRenderingData.minColumn, lineRenderingData.maxColumn);
+		const renderLineInput: RenderLineInput = new RenderLineInput(true, true, modifiedLine, lineRenderingData.continuesWithWrappedLine,
+			lineRenderingData.isBasicASCII, lineRenderingData.containsRTL, 0, lineRenderingData.tokens, actualInlineDecorations, lineRenderingData.tabSize,
+			lineRenderingData.startVisibleColumn, 1, 1, 1, 500, 'none', true, true, null);
+
+		const sb = createStringBuilder(2000);
+		renderViewLine(renderLineInput, sb);
+
+		let newLine;
+		if (_ttPolicy) {
+			newLine = _ttPolicy.createHTML(sb.build() as string);
+		} else {
+			newLine = sb.build();
+		}
+
+		const lineHTMLNode = document.createElement('div');
+		lineHTMLNode.style.paddingLeft = this._editor.getLayoutInfo().contentLeft - this._editor.getLayoutInfo().lineNumbersLeft - this._editor.getLayoutInfo().lineNumbersWidth + 'px';
+		lineHTMLNode.style.float = 'left';
+		lineHTMLNode.style.width = this._editor.getLayoutInfo().width - this._editor.getLayoutInfo().contentLeft + 'px';
+		lineHTMLNode.style.backgroundColor = `var(--vscode-stickyScroll-background)`;
+		lineHTMLNode.innerHTML = newLine as string;
+
+		const lineNumberHTMLNode = document.createElement('div');
+		lineNumberHTMLNode.style.width = this._editor.getLayoutInfo().contentLeft.toString() + 'px';
+		lineNumberHTMLNode.style.backgroundColor = `var(--vscode-stickyScroll-background)`;
+		lineNumberHTMLNode.style.color = 'var(--vscode-editorLineNumber-foreground)';
+
+		const innerLineNumberHTML = document.createElement('div');
+		innerLineNumberHTML.innerText = this._lineNumber.toString();
+		innerLineNumberHTML.style.paddingLeft = this._editor.getLayoutInfo().lineNumbersLeft.toString() + 'px';
+		innerLineNumberHTML.style.width = this._editor.getLayoutInfo().lineNumbersWidth.toString() + 'px';
+		innerLineNumberHTML.style.backgroundColor = `var(--vscode-stickyScroll-background)`;
+		innerLineNumberHTML.style.textAlign = 'right';
+		innerLineNumberHTML.style.float = 'left';
+		lineNumberHTMLNode.appendChild(innerLineNumberHTML);
+
+		lineHTMLNode.onclick = e => {
```

Comment:
This listener is leaking and would need to be captured in a disposable. You can use `addDisposableListener` from `vs/base/browser/dom.ts`.